src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.ts

```diff
@@ -0,0 +1,77 @@
+/** Angular Imports */
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { MatDialog } from '@angular/material';
+import { DatePipe } from '@angular/common';
+
+/** Custom Services */
+import { RecurringDepositsService } from '../recurring-deposits.service';
+
+/** Custom Buttons Configuration */
+import { RecurringDepositsButtonsConfiguration } from './recurring-deposits-buttons.config';
+
+/**
+ * RecurringDeposits Account View Component
+ */
+@Component({
+  selector: 'mifosx-recurring-deposits-account-view',
+  templateUrl: './recurring-deposits-account-view.component.html',
+  styleUrls: ['./recurring-deposits-account-view.component.scss']
+})
+export class RecurringDepositsAccountViewComponent implements OnInit {
+
+  /** RecurringDeposits Account Data */
+  recurringDepositsAccountData: any;
+  buttonConfig: RecurringDepositsButtonsConfiguration;
+  charges: any;
+
+  /**
+   * Fetches recurringDeposits account data from `resolve`
+   * @param {ActivatedRoute} route Activated Route
+   * @param {Router} router Router
+   * @param {RecurringDepositsService} recurringDepositsService RecurringDeposits Service
+   */
+  constructor(private route: ActivatedRoute,
+              private router: Router,
+              private datePipe: DatePipe,
+              private recurringDepositsService: RecurringDepositsService,
+              public dialog: MatDialog) {
+    this.route.data.subscribe((data: { recurringDepositsAccountData: any }) => {
+      this.recurringDepositsAccountData = data.recurringDepositsAccountData;
+      this.charges = this.recurringDepositsAccountData.charges;
+    });
+  }
+
+  ngOnInit() {
+    this.setConditionalButtons();
+  }
+
+  formatFrequencyType(val: string) {
+    const type = val.split(',')[-1];
+    return type.charAt(0).toUpperCase() + type.slice(1);
+  }
```

Comment:
use titlecase pipe, also dont use custom functions just to modify strings in the view

Reply:
presently this function isn't used, need some further work on it.